src/tree.c

```diff
@@ -797,14 +803,33 @@ int git_treebuilder_remove(git_treebuilder *bld, const char *filename)
 
 int git_treebuilder_write(git_oid *oid, git_treebuilder *bld)
 {
+	int error;
+	size_t entrycount;
+	git_buf buffer = GIT_BUF_INIT;
+
+	entrycount = git_strmap_num_entries(bld->map);
+
+	/* Grow the buffer beforehand to an estimated size */
+	error = git_buf_grow(&buffer, entrycount * 72);
+	if (error >= 0)
+		error = git_treebuilder_write_buffered(oid, bld, &buffer);
+
+	git_buf_free(&buffer);
+	return error;
+}
+
+int git_treebuilder_write_buffered(git_oid *oid, git_treebuilder *bld, git_buf *shared_buf)
+{
 	int error = 0;
 	size_t i, entrycount;
-	git_buf tree = GIT_BUF_INIT;
```

Comment:
The rename from `tree` to `shared_buf` adds noise to the change without aiding to the readability of the code. It's _still_ the buffer where we store the tree, it just happens to be an input.